File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/java/Simple.java
import org.jspecify.annotations.*;

public class Simple {
    public static class Base {}
    public static class Derived extends Base {
        void foo() {}
    }

    @Nullable public Simple.Derived field = null;

    @Nullable
    public Simple.Derived foo(Derived x, @NullnessUnspecified Base y) {
        return null;
    }

    static public void main(Simple a) {
        a.foo(new Derived(), null).foo();
        a.foo(null, new Derived()).foo();

        a.field.foo();
    }
}

File: compiler/testData/foreignAnnotationsJava8/tests/jspecify/simple.kt
// !DIAGNOSTICS: -UNUSED_VARIABLE -UNUSED_PARAMETER
// FILE: A.java

import org.jspecify.annotations.*;

public class A {
    @Nullable public String field = null;

    @Nullable
    public String foo(String x, @NullnessUnknown CharSequence y) {
        return "";
    }

}

// FILE: main.kt

fun main(a: A) {
    a.foo("", null)?.length
    a.foo("", null)<!UNSAFE_CALL!>.<!>length

    a.field?.length
    a.field<!UNSAFE_CALL!>.<!>length
}
